include/aspect/newton.h

```diff
@@ -61,6 +61,16 @@ namespace aspect
    * A Class which can declare and parse parameters and creates
    * material model outputs for the Newton solver.
    */
+
+  enum class PreconditionerStabilization
```

Comment:
The comment above this enum is still about the `NewtonHandler` class.

I'd say move these `enum`s *into* the `NewtonHandler` class. Also add documentation for both.